addon/services/permissions.ts

```diff
@@ -33,37 +41,29 @@ export default class PermissionsService extends Service {
     sendEvent(this, EVENTS.ROUTE_PERMISSIONS_CHANGED);
   }
 
-  cacheInitialTransition() {
-    addListener(
-      this.routerService,
-      'routeWillChange',
-      this,
-      (transition) => {
-        this.initialTransition = transition;
-      },
-      true
-    );
+  cacheInitialTransition(): void {
+    this.routerService.on('routeWillChange', (transition: Transition) => {
+      this.initialTransition = transition;
+    });
 
-    addListener(
-      this.routerService,
-      'routeDidChange',
-      this,
-      () => {
-        this.initialTransition = null;
-      },
-      true
-    );
+    this.routerService.on('routeDidChange', () => {
```

Comment:
I had the same addListener type issue here. People in the discord suggested the use of `routerService.on` instead. It's not the real fix though. Should I stick with addListener here as well?